core/src/test/scala/integration/kafka/api/PlaintextEndToEndAuthorizationTest.scala

```diff
@@ -76,14 +78,10 @@ class PlaintextEndToEndAuthorizationTest extends EndToEndAuthorizationTest {
   @Test
   def testListenerName() {
     // To check the client listener name, establish a session on the server by sending any request eg sendRecords
-    try {
-      sendRecords(1, tp)
-      fail("Should have thrown a TopicAuthorizationException")
-    } catch {
-      case tae: TopicAuthorizationException => //expected
-    }
-    assertEquals("CLIENT", PlaintextEndToEndAuthorizationTest.clientListenerName)
-    assertEquals("SERVER", PlaintextEndToEndAuthorizationTest.serverListenerName)
+    intercept[TopicAuthorizationException](sendRecords(1, tp))
+
+    assertEquals("CLIENT", PlaintextEndToEndAuthorizationTest.clientListenerName.get)
+    assertEquals("SERVER", PlaintextEndToEndAuthorizationTest.serverListenerName.get)
```

Comment:
The right way to do this is `assertEquals(Some("SERVER"), PlaintextEndToEndAuthorizationTest.serverListenerName)` so that one gets a reasonable error if the Option is None.

Reply:
Good point, fixed